src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -0,0 +1,190 @@
+package com.commercetools.sync.categories;
+
+import com.commercetools.sync.categories.helpers.CategorySyncOptions;
+import com.commercetools.sync.categories.helpers.CategorySyncStatistics;
+import com.commercetools.sync.categories.utils.CategorySyncUtils;
+import com.commercetools.sync.commons.Sync;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.categories.CategoryDraft;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.models.SphereException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+public class CategorySync implements Sync {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CategorySync.class);
+    private final CategorySyncOptions syncOptions;
+    private final CategorySyncStatistics statistics;
+
+    public CategorySync(@Nonnull final CategorySyncOptions syncOptions) {
+        this.syncOptions = syncOptions;
+        statistics = new CategorySyncStatistics();
+    }
+
+    /**
+     * Traverses a {@link List} of {@link CategoryDraft} objects and tries to fetch a category, from the CTP project with
+     * the configuration stored in the {@code syncOptions} instance of this class, using the external id. If a category exists,
+     * this category is synced to be the same as the new category draft in this list. If no category exist with such external id,
+     * a new category, identical to this new category draft  is created.
+     * <p>
+     * The {@code statistics} instance is updated accordingly to whether the CTP request was carried out successfully or not.
+     * If an exception was thrown on executing the request to CTP,
+     * the optional error callback specified in the {@code syncOptions} is called.
+     *
+     * @param categoryDrafts the list of new category drafts to sync to the CTP project.
+     * @param <T>            the type of the resource draft. In this case CategoryDraft.
+     */
+    @Override
+    public <T> void syncDrafts(@Nonnull final List<T> categoryDrafts) {
+        LOGGER.info(format("About to sync %d category drafts into CTP project with key '%s'."
+                , categoryDrafts.size(), getSyncOptions().getClientConfig().getProjectKey()));
+        for (int i = 0; i < categoryDrafts.size(); i++) {
+            try {
+                final CategoryDraft categoryDraft = (CategoryDraft) categoryDrafts.get(i);
+                if (categoryDraft != null) {
+                    getStatistics().incrementProcessed();
+                    final String externalId = categoryDraft.getExternalId();
+                    if (isNotBlank(externalId)) {
```

Comment:
Any customer has an `id` to uniquely identify their categories in their external pm system and that's what should be used in the `externalId` in CTP. That's what the https://github.com/sphereio/sphere-category-sync does as well. However, I see no problem to sync by slug as well. But could be added later as a feature.